agents/agents-features/agents-features-opentelemetry/src/jvmMain/kotlin/ai/koog/agents/features/opentelemetry/feature/OpenTelemetry.kt

```diff
@@ -413,55 +482,55 @@ public class OpenTelemetry {
 
             //region Tool Call
 
-            pipeline.interceptToolCallStarting(this) { eventContext ->
+            pipeline.interceptToolCallStarting(this) intercept@{ eventContext ->
                 logger.debug { "Execute OpenTelemetry tool call handler" }
 
                 // Get current NodeExecuteSpan
-                val agentRunInfoElement = getAgentRunInfoElement()
-                    ?: return@interceptToolCallStarting
-
-                val nodeInfoElement = getNodeInfoElement()
-                    ?: error("Unable to create tool call span due to missing node info in context")
+                val agentRunInfoElement = getAgentRunInfoElementCatching() ?: return@intercept
+                val nodeInfoElement = getNodeInfoElementCatching() ?: return@intercept
 
                 val nodeExecutionSpanId = NodeExecuteSpan.createId(
                     agentId = agentRunInfoElement.agentId,
                     runId = agentRunInfoElement.runId,
-                    nodeName = nodeInfoElement.nodeName
+                    nodeId = nodeInfoElement.id,
+                    nodeName = nodeInfoElement.name,
                 )
 
                 val nodeExecuteSpan = spanProcessor.getSpanCatching<NodeExecuteSpan>(nodeExecutionSpanId)
-                    ?: return@interceptToolCallStarting
+                    ?: return@intercept
 
+                @Suppress("UNCHECKED_CAST")
                 val executeToolSpan = ExecuteToolSpan(
                     parent = nodeExecuteSpan,
-                    tool = eventContext.tool,
-                    toolArgs = eventContext.toolArgs,
+                    toolName = eventContext.tool.name,
+                    toolDescription = eventContext.tool.description,
+                    toolArgs = (eventContext.tool as Tool<Any?, Any?>).encodeArgsToString(eventContext.toolArgs),
```

Comment:
Here and in other similar places: please let's be mindful of unchecked casts and let's try to encapsulate them,  providing safety checks, as much as possible. In this particular case, let's add `encodeArgsUnsafe` to the `Tool`, similar to the already existing `encodeResultUnsafe`. I plan to introduce actual safety checks and descriptive exceptions later, to help avoid non-descriptive `ClassCastException`. It would be easier to do if we narrow unsafe cast to a single place